### PR TITLE
Lint Treon + Travis

### PR DESCRIPTION
**Pull request by f4lco**
This sets up Pylint with exception of the missing docstring rule. I added the promised `requirements-dev.txt`, as well as basic CI integration and badge. I suppose you need to go to http://travis-ci.org and flip the toggle to enable Travis for treon.

### CODE REVIEW

File: treon/treon.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
+# pylint: disable=line-too-long
```
Comment (amit1rrr) [inline treon/treon.py]: Should we disable this for entire repo?

File: treon/task.py [hunk 2]
```diff
@@ -27,14 +27,14 @@ def run_tests(self):
                 result += console_output
 
             LOG.info(result)
-        except Exception:
+        except Exception:  # pylint: disable=broad-except
```
Comment (amit1rrr) [inline treon/task.py]: Should we disable this for entire repo?

File: README.md [hunk 3]
```diff
@@ -107,6 +110,8 @@ $ pip list --local | grep treon # should point to the source tree
 treon                0.1.2                /workspace/treon
 ```
 
+Please refer to the `Makefile` for supplementary development tasks, such as linting treon's source code.
```
Comment (amit1rrr) [inline README.md]: Can we please specify how to run the linter? People like me are not very familiar with makefiles :)

Also good to add note for future contributors to run linter before submitting PR.

**Comment by amit1rrr** [comment]
Thanks! Just left some minor comments. Also flicked the switch on Travis CI.

File: treon/treon.py (hunk 1 above)
Comment (f4lco) [reply treon/treon.py]: Actually, I think that Pylint has a good default of a maximum 100 characters per line, and longer lines are unreasonable (except for this one docstring).

File: treon/task.py (hunk 2 above)
Comment (f4lco) [reply treon/task.py]: No, better we fix it. There is a number of problems with this branch:
* The underlying method raises `CellExecutionError`, which is the proper exception type.
* Said exception already ships a stringified traceback of the cell, which we can simply use. The traceback of treon itself (as formatted by `traceback.format_exc`) is not relevant at this point.
*  The template from `error_string` method adds extra whitespace in the beginning of the traceback, despite `textwrap.dedent`:

```
ERROR in testing nb/Testi.ipynb
            Traceback (most recent call last):
  File "/redacted/treon/treon/task.py", line 23, in run_tests
etc.
```

I'll include a separate commit addressing these concerns.

File: README.md (hunk 3 above)
Comment (f4lco) [reply README.md]: OK no problem 👍

File: treon/task.py (hunk 2 above)
Comment (amit1rrr) [reply treon/task.py]: Sounds good. Feel free to make a separate PR for that one.

File: README.md (hunk 3 above)
Comment (amit1rrr) [reply README.md]: I am happy to merge the PR once you update the Readme.

**Comment by f4lco** [comment]
Thank god the builds were green 😄 And the badge appears to be working: [Build Status](https://travis-ci.org/f4lco/treon)

**Comment by amit1rrr** [comment]
Nice, it rarely builds successfully the first time 🙃

**Pull request merged**